src/resources/i2s_esp32.cc

```diff
@@ -181,11 +216,20 @@ IRAM_ATTR static bool channel_read_handler(i2s_chan_handle_t handle,
   return channel_send(resource);
 }
 
-IRAM_ATTR static bool channel_error_handler(i2s_chan_handle_t handle,
-                                  i2s_event_data_t* event,
-                                  void* user_ctx) {
+IRAM_ATTR static bool channel_overrun_error_handler(i2s_chan_handle_t handle,
+                                                     i2s_event_data_t* event,
+                                                     void* user_ctx) {
   auto resource = reinterpret_cast<I2sResource*>(user_ctx);
-  resource->inc_errors();
+  resource->inc_errors_overrun();
+  resource->adjust_pending_event(kErrorState);
```

Comment:
Also mark `I2sResource::adjust_pending_event` and `EventQueueResource::queue` with ATTR_IRAM.